test/test_cuda.py

```diff
@@ -2918,10 +2918,9 @@ def generate_tests():
                 else:
                     name_inner = name
 
-                if t != torch.HalfTensor and not hasattr(tensor, name_inner):
-                    # torch.HalfTensor doesn't support most operations,
-                    # but we use torch.FloatTensor as cpu baseline
+                if not hasattr(tensor, name_inner):
```

Comment:
Oops,  I allowed test for `cuda.HalfTensor`,  in which case, the above condition allowed `mean_` which crashed the test. 

Skipped over me, that `cuda.HalfTensor` was not an integral type.

However reading at the comment, I feel it should be 
```python
if t == torch.HalfTensor and and not hasattr(tensor, name_inner):
    continue
``` 

Like in this case the current condition allowed, `mean_` inplace for HalfTensor

I believe it would be better to check `hasattr` for all types.
Your thought?

Reply:
Have you undone the changes, kept `HalfTensor.mean` in `cuda_ignore` as this looks to be out-of-scope of this PR. However do let me know your thoughts regarding the above comments.

Thank You.